staging/src/k8s.io/apimachinery/pkg/util/sets/BUILD

```diff
@@ -38,7 +38,7 @@ go_genrule(
     ],
     cmd = """
 $(location //vendor/k8s.io/code-generator/cmd/set-gen) \
-    --input-dirs k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/sets/types \
+    --input-dirs k8s.io/apimachinery/pkg/util/sets/types \
```

Comment:
this seems a bit surprising

Reply:
I mentioned this above. Changes related to gopath are expected. We fixed go_genrule in kubernetes/repo-infra#122 to set things up properly.

Reply:
In GOPATH I would still expect this to be `k8s.io/kubernetes/{vendor,staging/...}/apimachinery/` to refer to the one in this directory tree as opposed to the staged copy at some other version.

Reply:
It doesn't really. It has worked so far because we don't have a go.mod in the build sandbox. It wouldn't work if the go genrule commands were run unsandboxed. These commands both fail in the KUBE_ROOT:
```
go list k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/sets
go list -mod=vendor k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/sets
```
The change kubernetes/repo-infra#122 was specifically meant to allow go_genrule invocations to match the invocations of the ./hack code-gen:

https://github.com/kubernetes/kubernetes/blob/c4fd09d80a3cbcb11d202b78987ad66ab4fd9904/hack/update-codegen.sh#L65-L69

```golang
$ (cd "${KUBE_ROOT}/staging/src"; find k8s.io/api -name types.go -print0 | xargs -0 -n1 dirname | sort)
k8s.io/api/admissionregistration/v1
k8s.io/api/admissionregistration/v1beta1
k8s.io/api/admission/v1
k8s.io/api/admission/v1beta1
k8s.io/api/apps/v1
k8s.io/api/apps/v1beta1
k8s.io/api/apps/v1beta2
k8s.io/api/auditregistration/v1alpha1
k8s.io/api/authentication/v1
k8s.io/api/authentication/v1beta1
k8s.io/api/authorization/v1
k8s.io/api/authorization/v1beta1
...
```
Isn't this better?